integration-tests/jpa/src/main/java/org/apache/camel/quarkus/component/jpa/it/JpaResource.java

```diff
@@ -49,6 +56,51 @@
         return consumerTemplate.receiveBodyNoWait("jpa:" + Fruit.class.getName(), List.class);
     }
 
+    @Path("/get/{id}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Fruit getFruit(@PathParam("id") int id) {
+        return producerTemplate.requestBody("jpa:" + Fruit.class.getName() + "?findEntity=true", id, Fruit.class);
+    }
+
+    @Path("/get/query/{id}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @SuppressWarnings("unchecked")
+    public List<Fruit> getFruitByQuery(@PathParam("id") int id) {
+        return producerTemplate.requestBody("jpa:" + Fruit.class.getName() + "?query="
+                + "select f from " + Fruit.class.getName() + " f where f.id = " + id,
+                null, List.class);
+    }
+
+    @Path("/get/query/named/{id}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @SuppressWarnings("unchecked")
+    public List<Fruit> getFruitByNamedQuery(@PathParam("id") int id) {
+        Map<String, Object> queryParams = new HashMap<>();
+        queryParams.put("fruitId", id);
+        context.getRegistry().bind("parameters", queryParams);
```

Comment:
Shouldn't we add a comment here that this is not a good pattern to copy in a real application?

Reply:
I can rework it so that it gets bound to the registry via a producer method, which is a bit more 'real world'.

Reply:
No, this OK for a test, esp. if we do it only once. I just wanted that ppl. see that this not good for areal app.